packages/typed-openapi/src/map-openapi-endpoints.ts

```diff
@@ -88,15 +88,22 @@ export const mapOpenApiEndpoints = (doc: OpenAPIObject) => {
       // Make parameters optional if none of them are required
       if (params) {
         const t = createBoxFactory({}, ctx);
-        if (params.query && lists.query.length && lists.query.every((param) => !param.required)) {
-          if (!params.query) params.query = {};
-          params.query = t.reference("Partial", [t.object(params.query)]) as any;
-        }
-        if (params.path && lists.path.length && lists.path.every((param) => !param.required)) {
-          params.path = t.reference("Partial", [t.object(params.path)]) as any;
-        }
-        if (params.header && lists.header.length && lists.header.every((param) => !param.required)) {
-          params.header = t.reference("Partial", [t.object(params.header)]) as any;
+
+        let k: keyof typeof params;
+        for (k in params) {
+          if (k !== "body") {
```

Comment:
@astahmer I'm thinking it could also be changed to explicitly check for the `query`, `path` and `header` values instead in order to future-proof it against changes to the `params` object.

Reply:
yeah that might be easier to deal with if something needs to be done specifically on one of those key